wfe/wfe.go

```diff
@@ -329,6 +331,60 @@ func (wfe *WebFrontEndImpl) handleKey(
 	}
 }
 
+func (wfe *WebFrontEndImpl) handleCertStatusBySerial(
+	ctx context.Context,
+	response http.ResponseWriter,
+	request *http.Request) {
+
+	serialStr := strings.TrimPrefix(request.URL.Path, certStatusBySerial)
+	serial := big.NewInt(0)
+	if _, ok := serial.SetString(serialStr, 16); !ok {
+		response.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	var status string
+	var cert *core.Certificate
+	var rcert *core.RevokedCertificate
+	if rcert = wfe.db.GetRevokedCertificateBySerial(serial); rcert != nil {
+		status = "Revoked"
+		cert = rcert.Certificate
+	} else if cert = wfe.db.GetCertificateBySerial(serial); cert != nil {
+		status = "Valid"
+	}
+
+	if status == "" || cert == nil {
+		response.WriteHeader(http.StatusNotFound)
+		return
+	}
+	result := struct {
+		Status      string
+		Serial      string
+		Certificate string
+		Reason      *uint  `json:",omitempty"`
+		RevokedAt   string `json:",omitempty"`
+	}{
+		Status:      status,
+		Serial:      serial.Text(16),
+		Certificate: string(cert.PEM()),
+	}
+	if rcert != nil {
+		if rcert.Reason != nil {
+			result.Reason = rcert.Reason
+		}
+		result.RevokedAt = rcert.RevokedAt.UTC().String()
+	}
+
+	resultJSON, err := marshalIndent(result)
```

Comment:
I think this line and most of the logic that follows can be replaced with one call to the `webFrontEndImpl.writeJSONResponse` function, like:

```
_ = wfe.writeJSONResponse(response, http.StatusOK, result)
```

In this case I think it would be fair to throw away the potential error return, there isn't much to do with it.

Reply:
Changed. I've decided to keep the `http.StatusInternalServerError` return in case serialization fails. It's better to indicate some kind of error to the client, that makes handling errors for callers much easier. A empty 200/204 response (or whatever Go will send by default) makes debugging and client-side error handling more complex.

Reply:
Good points, that works for me :+1: